src/ConsumeContexts.jl

```diff
@@ -7,14 +7,17 @@ using ..ChunkedBase: set!, dec!
 export AbstractConsumeContext, SkipContext
 export setup_tasks!, consume!, task_done!, sync_tasks, cleanup
 
+# End users should subtype this to create custom consume contexts which are then
+# used in `parse_file_parallel` and `parse_file_serial`, to dispatch on their
+# `populate_result_buffer!` method.
 abstract type AbstractConsumeContext end
```

Comment:
I'd put this in a docstring, not a comment, since this is public and user-facing